### PR TITLE
SNOW-817091 async executions documentation

### PR DESCRIPTION
**Pull request by sfc-gh-knozderko**
### Description
Async executions documentation.

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -503,6 +503,63 @@ Note that because this method is not available in the generic `IDataReader` inte
...
+Another is asynchronous execution of the query by the server (you can recognize it by `InAsyncMode` containing method names, e. g. `ExecuteInAsyncMode`, `ExecuteAsyncInAsyncMode`).
+
+Example of synchronous code starting a query to be executed asynchronously on the server:
+```cs
+using (SnowflakeDbConnection conn = new SnowflakeDbConnection("account=testaccount;username=testusername;password=testpassword"))
+{
+      conn.Open();
+      SnowflakeDbCommand cmd = (SnowflakeDbCommand)conn.CreateCommand();
+      cmd.CommandText = "SELECT ...";
+      var queryId = cmd.ExecuteInAsyncMode();
```
Comment (sfc-gh-pfus) [inline README.md:522]: So it is synchronous or asynchronous? 515 lines says sync.

Comment (sfc-gh-knozderko) [reply README.md:522]: it is synchronous code in terms of C# but asynchronous in terms of server query execution

File: README.md [hunk 2]
```diff
@@ -503,6 +503,63 @@ Note that because this method is not available in the generic `IDataReader` inte
...
+Assert.IsFalse(conn.IsAnError(queryStatus)); // assuming that the query has not finished with error
+```
+or the same in an asynchronous code:
+```cs
+var queryStatus = await cmd.GetQueryStatusAsync(queryId, CancellationToken.None).ConfigureAwait(false);
+Assert.IsTrue(conn.IsStillRunning(queryStatus)); // assuming that the query is still running
+Assert.IsFalse(conn.IsAnError(queryStatus)); // assuming that the query has not finished with error
+```
+
+The following example shows how to get query results. The operation will repeatedly check the query status until the query is completed or timeout happended or reaching the maximal number of attempts. The synchronous code example:
```
Comment (sfc-gh-pbulawa) [inline README.md]: Could you please change `the maximal number of attempts` to `the maximum number of attempts`

Comment (sfc-gh-knozderko) [reply README.md]: done

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Pull request merged**